### PR TITLE
Update Lib to Android X

### PR DESCRIPTION
**Pull request by RobertKeazor**
Update Libs to AndroidX
Update Kotlin to 1.3.0

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -18,15 +18,15 @@ project.ext {
 }
 
 android {
-    compileSdkVersion project.ext.compileSdkVersion
-    buildToolsVersion project.ext.buildToolsVersion
+    compileSdkVersion 28
```
Comment (ujavid) [inline app/build.gradle]: Can you revert this since it's pulled from the variables

File: app/build.gradle [hunk 2]
```diff
@@ -40,17 +40,13 @@ android {
 }
 
 dependencies {
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
-    implementation project(":kombind")
+    implementation fileTree(include: ['*.jar'], dir: 'libs')
```
Comment (ujavid) [inline app/build.gradle]: Revert this line, not necessary

File: app/build.gradle [hunk 3]
```diff
@@ -40,17 +40,13 @@ android {
...
+    implementation project(':kombind')
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
-    implementation "com.android.support:support-v4:$support_library_version"
-    implementation "com.android.support:appcompat-v7:$support_library_version"
-    implementation "com.android.support:cardview-v7:$support_library_version"
-
-    //DataBinding
-    kapt "com.android.databinding:compiler:$android_plugin_version"
-
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (ujavid) [inline app/build.gradle]: Can you create ext variables?

File: app/build.gradle [hunk 4]
```diff
@@ -40,17 +40,13 @@ android {
...
+    implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'androidx.cardview:cardview:1.0.0'
     testImplementation "junit:junit:$junit_version"
-    androidTestImplementation "com.android.support.test:runner:$test_runner_version"
-    androidTestImplementation "com.android.support.test.espresso:espresso-core:$espresso_version"
+    androidTestImplementation 'androidx.test:runner:1.1.0'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.0'
```
Comment (ujavid) [inline app/build.gradle]: Same here for variables

File: app/src/main/res/layout/item_simpleitem.xml [hunk 5]
```diff
@@ -26,14 +26,18 @@
                 android:layout_marginStart="@dimen/spacing_large"
                 android:layout_marginEnd="0dp"
                 android:textStyle="bold"
-                android:text="@string/name"/>
+                android:text="@string/name"
+                android:layout_marginLeft="@dimen/spacing_large"
+                android:layout_marginRight="0dp" />
```
Comment (ujavid) [inline app/src/main/res/layout/item_simpleitem.xml]: Not needed

File: app/src/main/res/layout/item_simpleitem.xml [hunk 6]
```diff
@@ -26,14 +26,18 @@
...
             <TextView
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
                 android:layout_marginStart="@dimen/spacing_large"
                 android:layout_marginEnd="0dp"
                 android:text="@{item.name}"
-                tools:text="Name"/>
+                tools:text="Name"
+                android:layout_marginLeft="@dimen/spacing_large"
+                android:layout_marginRight="0dp" />
```
Comment (ujavid) [inline app/src/main/res/layout/item_simpleitem.xml]: Not needed

File: kombind/build.gradle [hunk 7]
```diff
@@ -39,20 +39,15 @@ android {
 
 dependencies {
     //Android Dependencies
-    api fileTree(dir: 'libs', include: ['*.jar'])
+    api fileTree(include: ['*.jar'], dir: 'libs')
```
Comment (ujavid) [inline kombind/build.gradle:44]: Unnecessary change

File: kombind/build.gradle [hunk 8]
```diff
@@ -39,20 +39,15 @@ android {
...
     api "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
-    api "com.android.support:appcompat-v7:$support_library_version"
-    api "com.android.support:design:$support_library_version"
-
+    api 'androidx.appcompat:appcompat:1.0.2'
```
Comment (ujavid) [inline kombind/build.gradle]: Can you create variables for this and below?

**Review by ujavid — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/item_simpleitem.xml (hunk 5 above)
Comment (RobertKeazor) [reply app/src/main/res/layout/item_simpleitem.xml]: its needed because of the minSdkLevel. Start and End are only supported in later versions

Comment (ujavid) [reply app/src/main/res/layout/item_simpleitem.xml]: In that case can you move them under start and end?

File: kombind/build.gradle (hunk 7 above)
Comment (RobertKeazor) [reply kombind/build.gradle:44]: The IDE migration tool switched it around.

File: app/src/main/res/layout/item_simpleitem.xml [hunk 9]
```diff
@@ -23,31 +23,37 @@
...
-                android:layout_marginStart="0dp"
-                android:layout_marginEnd="@dimen/spacing_normal"
                 android:padding="@dimen/spacing_normal"
                 android:background="?attr/selectableItemBackground"
                 android:textColor="@color/colorPrimary"
                 android:textStyle="bold"
                 android:text="@string/delete"
-                android:onClick="@{(view) -> handler.onDeleteClick(item.id)}"/>
+                android:onClick="@{(view) -> handler.onDeleteClick(item.id)}"
+                android:layout_marginStart="0dp"
```
Comment (ujavid) [inline app/src/main/res/layout/item_simpleitem.xml]: Can you move this back and put the left and right underneath it?

**Review by ujavid — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/item_simpleitem.xml (hunk 6 above)
Comment (ujavid) [reply app/src/main/res/layout/item_simpleitem.xml]: Move under start and end

File: app/build.gradle [hunk 10]
```diff
@@ -40,17 +42,13 @@ android {
...
-    implementation project(":kombind")
+    implementation fileTree(include: ['*.jar'], dir: 'libs')
+    implementation project(':kombind')
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
-    implementation "com.android.support:support-v4:$support_library_version"
-    implementation "com.android.support:appcompat-v7:$support_library_version"
-    implementation "com.android.support:cardview-v7:$support_library_version"
-
-    //DataBinding
-    kapt "com.android.databinding:compiler:$android_plugin_version"
```
Comment (ujavid) [inline app/build.gradle:51]: should the be removed?

File: kombind/build.gradle [hunk 11]
```diff
@@ -39,20 +41,15 @@ android {
...
-    api "com.android.support:appcompat-v7:$support_library_version"
-    api "com.android.support:design:$support_library_version"
-
+    api "androidx.appcompat:appcompat:$androidx_appcompat"
+    api "com.google.android.material:material:$android_material"
     //Lifecycle dependency
-    api "android.arch.lifecycle:extensions:$lifecycle_version"
-
-    //DataBinding
-    kapt "com.android.databinding:compiler:$android_plugin_version"
```
Comment (ujavid) [inline kombind/build.gradle:51]: same here, shouldn't this be included?

File: app/build.gradle (hunk 10 above)
Comment (RobertKeazor) [reply app/build.gradle:51]: They are not needed anymore. the plugin will inject it

**Review by ujavid — APPROVED** [review]
